app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,267 @@
+        self.modelsets_lim = execution_params.get('modelsets_lim', 3)
+
+    # make sure to gitignore model caches, because the model needs to be compiled separately
+    # on each machine - it is system-specific C++ code
+    def get_stan_model_cache(self, model_code: str, model_name: str = 'anon_model', **kwargs: Dict) -> pystan.StanModel:
+        """Use just as you would `pystan.StanModel`"""
+
+        # Create filepath of cached model
+        code_hash = md5(model_code.encode('ascii')).hexdigest()
+        cache_fn = f'stanmodelcache-{model_name}-{code_hash}.pkl'
+
+        # Try to load cached model
+        try:
+            cached_model = pickle.load(open(cache_fn, 'rb'))
+            print(f"Using cached StanModel at filepath {cache_fn}")
+
+        # Otherwise create the model and cache it
+        except FileNotFoundError:
+            cached_model = pystan.StanModel(model_code=model_code, model_name=model_name, **kwargs)
+            with open(cache_fn, 'wb') as f:
+                pickle.dump(cached_model, f)
+        return cached_model
+
+    def fit_one_pystan_model(self, model_params: Dict) -> Tuple:
+        if model_params['n_sp' < 10] or model_params['n_se' < 10]  \
+                or model_params['y_sp' < 5]  or model_params['y_se' < 5]:
+            adapt_delta = 0.99
+        else:
+            adapt_delta = 0.8
+        fit = self.TESTADJ_MODEL.sampling(data=model_params,
+                                          iter=self.n_iter,
+                                          chains=self.n_chains,
+                                          control={'adapt_delta': adapt_delta},
+                                          check_hmc_diagnostics=False)
+
+        summary = fit.summary()
+        summary_df = pd.DataFrame(data=summary['summary'],
+                                  index=summary['summary_rownames'],
+                                  columns=summary['summary_colnames'])
+
+        samples = fit.extract(pars='prev', permuted=True)['prev']
+
+        summary_df_parsed = summary_df.loc['prev', ['50%', '2.5%', '97.5%']]
+        summary_df_parsed['samples'] = samples
+        summary_df_parsed['fit'] = fit
+
+        diagnostics = pystan.diagnostics.check_hmc_diagnostics(fit)
+
+        satisfactory_model_found = all(diagnostics.values())
+        return summary_df_parsed, satisfactory_model_found
+
+    def pystan_adjust(self, model_params: Dict, execution_params: Dict = {}) -> Union[Tuple, pystan.StanModel]:
+        if self.n_replicates % 2 != 1:
+            raise ValueError(f'n_replicates must be odd')
+        credible_interval_size = execution_params.get('credible_interval_size', 0.95)
+
+        # Validate model_params using marshmallow
+        try:
+            ModelParamsSchema().load(model_params)
+        except ValidationError as err:
+            print("Error: ", err.messages)
+            return None, None, None
+
+        satisfactory_model_set_found = False
```

Comment:
We are going to running one model and seeing whether that single model is satisfactory, as opposed to running many model sets